components/locid/src/langid.rs

```diff
@@ -351,6 +351,18 @@ impl FromStr for LanguageIdentifier {
     }
 }
 
+impl Ord for LanguageIdentifier {
```

Comment:
Issue: we can't agree on what the behavior of `impl Ord for LanguageIdentifier` should be, so we don't want any impl of it. What things in ICU4X require it?

Reply:
These are the list of errors that I am running into if there is no Ord impl for LanguageIdentifier. This is probably not complete and most if not all of these errors are related to LiteMap.

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/calendar/japanese.rs#L60

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/decimal/mod.rs#L82

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/list/mod.rs#L42

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/locale_canonicalizer/aliases.rs#L79

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/plurals/mod.rs#L61

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/ca.rs#L165

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/likely_subtags.rs#L16

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/list_patterns.rs#L52

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/numbers.rs#L110

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/parent_locales.rs#L16

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/time_zones/time_zone_names.rs#L164

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/time_zones/time_zone_names.rs#L164

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/datetime/mod.rs#L89